flake.lock

```diff
@@ -0,0 +1,104 @@
+{
+  "nodes": {
+    "fenix": {
+      "inputs": {
+        "nixpkgs": [
+          "nixpkgs"
+        ],
+        "rust-analyzer-src": "rust-analyzer-src"
```

Comment:
Any idea why rust-analyzer is in the lock file, but isn't mentioned anywhere else in the nix files?

Reply:
I didn't know before but I do now.  The `flake.lock` locks transitive dependencies as well.  If you look at the structure here, it becomes apparent that `rust-analyzer-src` is an input to `fenix` which is an input to our flake.  This part of the lock file doesn't actually lock the version but down around line 84 there is another entry for `rust-analyzer-src` which does pin a version.

I also found https://github.com/NixOS/nix/issues/5433 which suggests the need to be a bit careful about updating dependencies with respective to transitive dependencies.

Also, maybe not directly related, but note that we make fenix and naersk flakes follow our pinned version of nixpkgs so we don't have a transitive dependency on _their_ pinned versions of nixpkgs (but this does mean we're potentially using them with a different version of nixpkgs than the developers or other users - however, the docs recommend this pattern).

We could also directly pin rust-analyzer-src as a direct input and then make fenix follow our pin.  I won't for now because we're not actually using rust-analyzer-src directly for anything (at least I'm not :slightly_smiling_face: ).